src/Fmt_ast.ml

```diff
@@ -894,6 +894,113 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
   update_config_maybe_disabled c ppat_loc ppat_attributes
   @@ fun c ->
   let parens = match parens with Some b -> b | None -> parenze_pat xpat in
+  let width xpat = String.length (Cmts.preserve (fmt_pattern c) xpat) in
+  let fmt_op_args op_args =
```

Comment:
This function is very similar with the one in `fmt_expression`. Would it be possible to share some parts ? (eg. the ones that implement formatting options)

Reply:
Should be factorized with the new `fmt_infix_op_args` option